app/services/container_dashboard_service.rb

```diff
@@ -209,37 +263,54 @@ def daily_network_metrics
       daily_network_metrics[day] += m.net_usage_rate_average if m.net_usage_rate_average.present?
     end
 
-    if daily_network_metrics.any?
-      {
-        :xData => daily_network_metrics.keys,
-        :yData => daily_network_metrics.values.map(&:round)
-      }
-    end
+    trend_data(daily_network_metrics) if daily_network_metrics.any?
   end
 
-  def fill_daily_pod_metrics(metrics, pod_create_trend, pod_delete_trend)
-    metrics.each do |m|
-      timestamp = m.timestamp.strftime("%Y-%m-%d")
+  def fill_pod_metrics(m, time, pod_create_trend, pod_delete_trend)
+    pod_create_trend[time] += m.stat_container_group_create_rate if m.stat_container_group_create_rate.present?
+    pod_delete_trend[time] += m.stat_container_group_delete_rate if m.stat_container_group_delete_rate.present?
+  end
 
-      pod_create_trend[timestamp] += m.stat_container_group_create_rate if m.stat_container_group_create_rate.present?
-      pod_delete_trend[timestamp] += m.stat_container_group_delete_rate if m.stat_container_group_delete_rate.present?
+  def hourly_pod_metrics
+    hourly_pod_create_trend = Hash.new(0)
+    hourly_pod_delete_trend = Hash.new(0)
+    hourly_provider_metrics.each do |m|
+      hour = m.timestamp.beginning_of_hour.utc
+      fill_pod_metrics(m, hour, hourly_pod_create_trend, hourly_pod_delete_trend)
     end
+
+    create_delete_data(hourly_pod_create_trend, hourly_pod_delete_trend) if hourly_pod_create_trend.any?
   end
 
   def daily_pod_metrics
     daily_pod_create_trend = Hash.new(0)
     daily_pod_delete_trend = Hash.new(0)
 
-    fill_daily_pod_metrics(daily_provider_metrics,
-                           daily_pod_create_trend, daily_pod_delete_trend)
+    daily_provider_metrics.each do |m|
+      date = m.timestamp.strftime("%Y-%m-%d")
+      fill_pod_metrics(m, date, daily_pod_create_trend, daily_pod_delete_trend)
+    end
 
-    if daily_pod_create_trend.any?
-      {
-        :xData    => daily_pod_create_trend.keys,
-        :yCreated => daily_pod_create_trend.values.map(&:round),
-        :yDeleted => daily_pod_delete_trend.values.map(&:round)
-      }
+    create_delete_data(daily_pod_create_trend, daily_pod_delete_trend) if daily_pod_create_trend.any?
+  end
+
+  def create_delete_data(create_trend, delete_trend)
+    {
+      :xData    => create_trend.keys,
+      :yCreated => create_trend.values.map(&:round),
+      :yDeleted => delete_trend.values.map(&:round)
+    }
+  end
+
+  def hourly_image_metrics
+    hourly_image_metrics = Hash.new(0)
+    hourly_provider_metrics.each do |m|
+      hour = m.timestamp.beginning_of_hour.utc
+      hourly_image_metrics[hour] +=
+        m.stat_container_image_registration_rate if m.stat_container_image_registration_rate.present?
     end
+
+    trend_data(hourly_image_metrics) if hourly_image_metrics.any?
```

Comment:
@kbrock just wondering, would you have substituted the `any?` calls for something else?